ksqldb-common/src/main/java/io/confluent/ksql/schema/ksql/SystemColumns.java

```diff
@@ -15,40 +15,68 @@
 
 package io.confluent.ksql.schema.ksql;
 
+import com.google.common.collect.ImmutableMap;
 import com.google.common.collect.ImmutableSet;
 import io.confluent.ksql.name.ColumnName;
 import io.confluent.ksql.schema.ksql.types.SqlType;
 import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import java.util.HashMap;
+import java.util.Map;
 import java.util.Set;
 
 public final class SystemColumns {
 
   public static final ColumnName ROWKEY_NAME = ColumnName.of("ROWKEY");
-
   public static final ColumnName ROWTIME_NAME = ColumnName.of("ROWTIME");
   public static final SqlType ROWTIME_TYPE = SqlTypes.BIGINT;
 
+  public static final ColumnName ROWOFFSET_NAME = ColumnName.of("ROWOFFSET");
+  public static final SqlType ROWOFFSET_TYPE = SqlTypes.BIGINT;
+
+  public static final ColumnName ROWPARTITION_NAME = ColumnName.of("ROWPARTITION");
+  public static final SqlType ROWPARTITION_TYPE = SqlTypes.INTEGER;
+
   public static final ColumnName WINDOWSTART_NAME = ColumnName.of("WINDOWSTART");
   public static final ColumnName WINDOWEND_NAME = ColumnName.of("WINDOWEND");
   public static final SqlType WINDOWBOUND_TYPE = SqlTypes.BIGINT;
 
   public static final int LEGACY_PSEUDOCOLUMN_VERSION_NUMBER = 0;
-  public static final int CURRENT_PSEUDOCOLUMN_VERSION_NUMBER = 5;
+  public static final int CURRENT_PSEUDOCOLUMN_VERSION_NUMBER = 0;
+
+  public static Set<ColumnName> getPseudoColumnsFromVersion(final int version) {
+    return versionedPseudoColumns.get(version);
```

Comment:
We need to handle the case where an invalid version number is passed. (Throw an informative exception.)